### PR TITLE
js: Add GetMsg API to JetStreamManager

### PR DESCRIPTION
**Pull request by wallyqs**
Signed-off-by: Waldemar Quevedo <wally@synadia.com>

### CODE REVIEW

File: jsm.go [hunk 1]
```diff
@@ -462,6 +465,57 @@ func (js *js) DeleteStream(name string) error {
...
+
+	var resp apiMsgGetResponse
+	if err := json.Unmarshal(r.Data, &resp); err != nil {
+		return nil, err
+	}
+	if resp.Error != nil {
+		return nil, errors.New(resp.Error.Description)
+	}
+
+	msg := resp.Message
```
Comment (derekcollison) [inline jsm.go:512]: Probably should return storedMsg. Need access to timestamp etc. Also below you forget headers.

Comment (derekcollison) [reply jsm.go:512]: Maybe call it StreamMsg or RawStreamMsg.

Comment (wallyqs) [reply jsm.go]: Updated and renamed it `StreamMsg`.

File: jsm.go [hunk 2]
```diff
@@ -462,6 +465,54 @@ func (js *js) DeleteStream(name string) error {
...
 
+type apiMsgGetRequest struct {
+	Seq uint64 `json:"seq"`
+}
+
+// StreamMsg is a raw message stored in JetStream.
+type StreamMsg struct {
+	Subject  string    `json:"subject"`
+	Sequence uint64    `json:"seq"`
+	Header   []byte    `json:"hdrs,omitempty"`
```
Comment (derekcollison) [inline jsm.go:476]: We should convert this if its there like a normal message no?

File: jsm.go (hunk 1 above)
Comment (wallyqs) [reply jsm.go:525]: Synced with @ColinSullivan1 and I think now that `RawStreamMsg` might be better avoid confusion with regular `Msg` so updated again. Also it is now decoding the headers from the fetched message and  converting from `[]byte` to `http.Header`.

File: jsm.go (hunk 2 above)
Comment (wallyqs) [reply jsm.go:488]: Updated to decode the header and use the http.Header type.

File: jsm.go (hunk 1 above)
Comment (derekcollison) [reply jsm.go:525]: ok

**Review by derekcollison — APPROVED** [review]
LGTM

**Pull request merged**